content/kubermatic/main/tutorials-howtos/storage/disable-csi-driver/_index.en.md

```diff
@@ -0,0 +1,22 @@
++++
+title = "Disable CSI driver addon on user clusters"
+date = 2023-10-30T14:43:42+05:30
+weight = 10
+
++++
+
+KKP installs the CSI drivers on user clusters which have the external CCM enabled, the CSI driver is installed as an [addon](https://github.com/kubermatic/kubermatic/tree/main/addons/csi) in the user cluster. However in some cases the KKP or the cluster admin might want to use some other storage driver in their data center/cluster, for this KKP provides an option to disable the CSI driver installation for the data center or specific clusters in the data center, if this option is enabled then KKP/Cluster admin is responsible to deploy & manage the storage solution on their own.
+
+### Disable CSI driver installation for data center
+To disable the CSI driver installation for all user cluster in a date center the admin needs to set ` disableCsiDriver: true` in the data center spec.
+
+User cluster created after enabling this option will not have the CSI addon installed & the cluster admins won't be able to over-ride and enable it in the cluster spec.
+
+This will not impact the clusters which were created prior to enabling this option & if desired it needs to be updated on the cluster spec.
+
+### Disable CSI driver installation for user cluster
+To disable the CSI driver installation for a user cluster, admin needs to set `disableCsiDriver: true` in the cluster spec, this is possible only if it is not disabled at the data center.
+
+User cluster created with `disableCsiDriver: true` will not have the CSI addon installed.
+
+User cluster which already have the CSI driver addon installed can also be updated to disable the CSI driver addon using the `disableCsiDriver: true` option but before doing this the cluster admin must ensure that none of the PVCs & PVs that belong to the storage class which has the CSI driver as the provisioner are in use, if there is any PVC which belongs to a storage class that uses the CSI driver KKP will not disable it. If for any reason the admin removes the storage class even if it has PVs which are in use & then disable the CSI driver then it will have undesired impact on the cluster so it is expected that admin are careful while making this change & do not remove the storage class before disabling the CSI driver.
```

Comment:
```suggestion
User clusters that already have the CSI driver addon installed can also be updated to disable the CSI driver addon using the `disableCsiDriver: true` option. However, before doing this the cluster admin must ensure that none of the PVCs & PVs that belong to the storage class that has the CSI driver as the provisioner are in use, if there is any PVC that belongs to a storage class that uses the CSI driver KKP will not disable it. If for any reason the admin removes the storage class even if it has PVs that are in use & then disables the CSI driver, it will have an undesired impact on the cluster. Hence, it is expected that the admin is careful while making this change & does not remove the storage class before disabling the CSI driver.
```